test/examples/ucp_hello_world.c

```diff
@@ -471,10 +482,13 @@ int main(int argc, char **argv)
             goto err_peer_addr;
         }
     }
-    close(oob_sock);
 
     ret = run_test(server, ucp_worker);
 
+    /* Make sure remote is disconnected before destroying local worker */
+    barrier(oob_sock);
```

Comment:
Is it UCX limitation? It is required to keep all MPI ranks alive (in regular application) before destroying workers. In case of client/server the client should wait until the server close the connection. I'm not sure it match TCP interchange practice (client could close connection and exits from one side).